src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -642,37 +645,14 @@ std::shared_ptr<MLModelWrapper> style_transfer::export_to_coreml(
 void style_transfer::import_from_custom_model(variant_map_type model_data,
                                               size_t version) {
   // Get relevant values from variant_map_type
-  auto model_iter = model_data.find("_model");
-  if (model_iter == model_data.end()) {
-    log_and_throw("The loaded turicreate model must contain '_model'!\n");
-  }
-  const flex_dict& model = variant_get_value<flex_dict>(model_iter->second);
-  model_data.erase(model_iter);
-
-  auto num_styles_iter = model_data.find("num_styles");
-  if (num_styles_iter == model_data.end()) {
-    log_and_throw("The loaded turicreate model must contain 'num_styles'!\n");
-  }
-  const size_t num_styles =
-      variant_get_value<flex_int>(num_styles_iter->second);
-  model_data.erase(num_styles_iter);
-
-  auto max_iterations_iter = model_data.find("max_iterations");
-  if (max_iterations_iter == model_data.end()) {
-    log_and_throw(
-        "The loaded turicreate model must contain 'max_iterations'!\n");
-  }
-  const size_t max_iterations =
-      variant_get_value<flex_int>(max_iterations_iter->second);
-  model_data.erase(max_iterations_iter);
-
-  auto model_type_iter = model_data.find("model");
-  if (model_type_iter == model_data.end()) {
-    log_and_throw("The loaded turicreate model must contain 'model_iter'!\n");
-  }
-  const std::string model_type =
-      variant_get_value<flex_string>(model_type_iter->second);
-  model_data.erase(model_type_iter);
+  const flex_dict& model =
+      read_opts<flex_dict>(model_data, "_model", /* delete item */ true);
```

Comment:
Why are we deleting these keys from the dictionary?

Reply:
I guess because we don't need them anymore so I delete them, is that bad?

Reply:
It's confusing, and therefore makes the code less readable and maintainable.

These will get deleted anyway when the dictionary goes out of scope.